modules/code-editor/src/views/full/FileNavigator.tsx

```diff
@@ -55,7 +55,10 @@ class FileNavigator extends React.Component<Props, State> {
   }
 
   refreshNodes = () => {
-    if (!this.props.files) {
+    if (this.state.nodes.length > 0 && this.props.filters && this.props.filters.filename === '') {
```

Comment:
This breaks new file creation, they are not displayed in the UI until you refresh. Not a good UX